app/src/main/java/com/alphawallet/app/ui/DappBrowserFragment.java

```diff
@@ -1176,13 +1179,39 @@ private void showChainChangeDialog(long callbackId, NetworkInfo newNetwork)
 
     private void handleSignMessage(Signable message)
     {
-        if (confirmationDialog == null || !confirmationDialog.isShowing())
+        if (message.getMessageType() == SignMessageType.SIGN_TYPED_DATA_V3 && message.getChainId() != activeNetwork.chainId)
+        {
```

Comment:
Does V4 have the chainId? As far as I'm aware the only difference between V3 and V4 is that V4 supports arrays, so we should check for SIGN TYPE DATA rather than SIGN TYPE DATA V3. Unless V3 also covers V4.